File: Algoritmos de ordenamiento/Lists/Lists.java
package Lists;

import java.util.ArrayList;

public class Lists {
    public static void main(String[] args) {
        ArrayList<Integer> lista_one = new ArrayList<Integer>();

        lista_one.add(2);
        lista_one.add(0);
        lista_one.add(3);

        System.out.println("Lista completa: " + lista_one);

        lista_one.set(0, 200);

        System.out.println("Primer elemento: " + lista_one.get(0));
        System.out.println("Segundo elemento: " + lista_one.get(1));

        boolean estaVació = lista_one.isEmpty();

        System.out.println("Método que comprueba si la lista esta vacía: " + estaVació);
        lista_one.remove(1);
        System.out.println("Lista después de eliminar el segundo elemento: " + lista_one);

        System.out.println("Limpiar lista");
        lista_one.clear();

        System.out.println("Longitud de la lista: " + lista_one.size());

    }
}

File: Algoritmos de ordenamiento/Lists/Lists.py
lista_one = list()


lista_one.append('Cristian')
lista_one.append('Jesus')
lista_one.append('Mario')

# Ordenamiento de burbuja
n = len(lista_one)
for i in range(n):
    for j in range(0, n-i-1):
        if lista_one[j] > lista_one[j+1]:
            lista_one[j], lista_one[j+1] = lista_one[j+1], lista_one[j]
print("Lista ordenada con burbuja:", lista_one)

# Ordenamiento por inserción
for i in range(1, len(lista_one)):
    key = lista_one[i]
    j = i-1
    while j >= 0 and key < lista_one[j]:
        lista_one[j+1] = lista_one[j]
        j -= 1
    lista_one[j+1] = key
print("Lista ordenada por inserción:", lista_one)

# Ordenamiento por selección
for i in range(len(lista_one)):
    min_idx = i
    for j in range(i+1, len(lista_one)):
        if lista_one[min_idx] > lista_one[j]:
            min_idx = j
    lista_one[i], lista_one[min_idx] = lista_one[min_idx], lista_one[i]
print("Lista ordenada por selección:", lista_one)

# Imprimir la lista
print("Lista completa:", lista_one)

# Acceder a elementos específicos de la lista
print("Primer elemento:", lista_one[0])
print("Segundo elemento:", lista_one[1])

# Eliminar un elemento de la lista
del lista_one[1]
print("Lista después de eliminar el segundo elemento:", lista_one)

# Longitud de la lista
print("Longitud de la lista:", len(lista_one))
